homeassistant/components/climate/__init__.py

```diff
@@ -449,6 +494,8 @@ class ClimateDevice(Entity):
     @property
     def state(self):
         """Return the current state."""
+        if self.is_on is False:
+            return STATE_OFF
```

Comment:
What happens if you turn on the device without current operation set? Unknown state? I would expect it to be `STATE_ON`.

Reply:
It might be an illegal state. It is up for the platform to handle this.

Reply:
How? By overwriting `state`? Then this change is kind of moot. Why not:
```py
def state(self):
    """Return the current state."""
    if self.is_on is False:
    return STATE_OFF
    if self.current_operation:
        return self.current_operation
    if self.is_on:
        return STATE_ON
    return None
```

Reply:
There is no such device. The only device supporting on/off is Sensibo (as of this PR).
And if it doesn't have current operation mode than it is in unknown state.

Reply:
Since we implement this on the component level, we should not only think about Sensibo. Either we have to separate the turn on/off service and state completely from operation mode, ie don't rely on operation mode being supported, or we should not implement the new service at all and let operation mode return on/off.

Reply:
I can add
```
if self.is_on:
        return STATE_ON
```
But currently there is no platform (including Sensibo) that will reach this code.

Reply:
Added

Reply:
The way I understood the combination you would have:
state: on(running, in_operation) or idle (off)
operation_mode: <whatever is supported>

Many thermostats will probably support "off" for operation_mode where they would ignore temperature changes.

"state" should be controlled by login in the thermostat, and you should not manually trigger it using a service (for most thermostats anyway).